lib/ibm/cloud/sdk_http/exceptions.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module IBM
+  module Cloud
+    module SDKHTTP
+      # Module to contain all custom exception classes.
+      module Exceptions
+        # An exception for http with a response attribute.
+        # @param response [IBM::Cloud::SDK::VPC::Response] The original response object.
+        # @param msg [String] A human readable message.
+        class ExceptionWithResponse < RuntimeError
+          def initialize(response, msg)
```

Comment:
Generally subclasses of an exception keep the message first and put the "extra" parameters last, otherwise you end up with an unexpected interface.

Reply:
True, but the important piece here is capturing the response. If you don't have a SDKResponse instance don't use this exception.

Reply:
Oh yeah I don't disagree, I'm just saying the order should be the other way around.

Reply:
Again, I'm talking syntax vs semantics... Semantically, it's fine and even works as is... syntactically I'm saying the convention is to keep message as the first parameter when creating custom Exceptions (where the message is still a parameter).

Separately is this technically a private class?  That is, is the only caller expected to be the `HttpStatusError`?  If so, then declaring it a private class in the yarddoc would probably be helpful.

Reply:
I use ExceptionWithResponse for other errors in SDKResponse class. I can't think of a use outside of SDKResponse, supposedly all the HTTP verification stuff is supposed to happen there. I leave it open for use within other BaseClasses that receive SDKResponses to raise and have the HTTPResponse object attached. Does that answer your question?

Reply:
Changed to `(msg, response)`